report/renderer/i18n.js

```diff
@@ -22,25 +22,47 @@ export class I18n {
     // When testing, use a locale with more exciting numeric formatting.
     if (locale === 'en-XA') locale = 'de';
 
-    this._numberDateLocale = locale;
-    this._numberFormatter = new Intl.NumberFormat(locale);
-    this._percentFormatter = new Intl.NumberFormat(locale, {style: 'percent'});
+    this._locale = locale;
     this._strings = strings;
   }
 
   get strings() {
     return this._strings;
   }
 
+  /**
+   * @param {number} number
+   * @param {number} granularity
+   * @param {Intl.NumberFormatOptions} opts
+   * @return {string}
+   */
+  _formatNumberWithGranularity(number, granularity, opts = {}) {
+    opts = {...opts};
+    const log10 = -Math.log10(granularity);
+    if (!Number.isFinite(log10) || (granularity > 1 && Math.floor(log10) !== log10)) {
+      throw new Error(`granularity of ${granularity} is invalid`);
+    }
+
+    if (granularity < 1) {
+      opts.minimumFractionDigits = opts.maximumFractionDigits = Math.ceil(log10);
+    }
+
+    number = Math.round(number / granularity) * granularity;
+
+    // Avoid displaying a negative value that rounds to zero as "0".
+    if (Object.is(number, -0)) number = 0;
```

Comment:
I could have done `if (number === 0) number = 0` but it was just too much.

Reply:
What if JS added `====` so `-0 ==== 0` would be false 🤪

Reply:
IEEE-754 requires that they compare as equal 🤓